src/main/kotlin/gov/nist/secauto/oscal/tools/server/OscalVerticle.kt

```diff
@@ -288,48 +292,49 @@ class OscalVerticle : CoroutineVerticle() {
             }
         }
     }
-        private fun mapMimeTypeToFormat(mimeType: String?, formatParam: String?): String {
-            // Check if a valid format parameter is provided
-            if (!formatParam.isNullOrBlank()) {
-                return when (formatParam.lowercase()) {
-                    "json" -> "JSON"
-                    "xml" -> "XML"
-                    "yaml" -> "YAML"
-                    else -> "JSON" // Default to JSON if an invalid format is provided
-                }
-            }
 
-            // If no valid format parameter, check the MIME type
-            mimeType?.lowercase()?.let {
-                return when {
-                    it.contains("json") -> "JSON"
-                    it.contains("xml") -> "XML"
-                    it.contains("yaml") -> "YAML"
-                    else -> "JSON" // Default to JSON if no valid MIME type is provided
-                }
+    private fun mapMimeTypeToFormat(mimeType: String?, formatParam: String?): String {
+        // Check if a valid format parameter is provided
+        if (!formatParam.isNullOrBlank()) {
+            return when (formatParam.lowercase()) {
+                "json" -> "JSON"
+                "xml" -> "XML"
+                "yaml" -> "YAML"
+                else -> "JSON" // Default to JSON if an invalid format is provided
```

Comment:
Blocking: I'd say at this point we should return an error and not presume JSON, even if that is easy to do.

Reply:
see new commit